core/store/src/trie/trie_storage.rs

```diff
@@ -184,31 +220,53 @@ impl TrieCachingStorage {
     fn inc_counter(&self) {
         self.counter.set(self.counter.get() + 1);
     }
+
+    /// If we are in the caching chunk state, put item to the chunk cache.
+    fn try_put_to_chunk_cache(&self, key: CryptoHash, value: Arc<[u8]>) {
+        if let TrieCacheState::CachingChunk = self.cache_state.borrow().get() {
+            self.chunk_cache.borrow_mut().insert(key, value);
+        };
+    }
```

Comment:
```suggestion
    fn try_put_to_chunk_cache(&self, key: CryptoHash, value: &Arc<[u8]>) {
        if let TrieCacheState::CachingChunk = self.cache_state.borrow().get() {
            self.chunk_cache.borrow_mut().insert(key, Arc::clone(value));
        };
    }
```

Экономия на спичках (saving matches), but let's avoid needless clone here.

I am also not super happy about naming `try_` means "failable", not optional. We use `maybe_` elsewhere, so we might as well put it here

Reply:
An alternative solution to the naming is to make sure there's only one call-site for the function, so that we can inline it:

```rust
    fn retrieve_raw_bytes(&self, hash: &CryptoHash) -> Result<Arc<[u8]>, StorageError> {
        // Try to get value from chunk cache containing free of charge nodes, otherwise increment the nodes counter.
        // let RawBytesWithCost { value, cost } = self.get_from_cache(hash);
        if let Some(val) = self.chunk_cache.borrow_mut().get(hash) {
            return Ok(val.clone());
        }

        // If full cost for retrieving the node must be charged, increment the counter.
        let val = match self.shard_cache.get(hash) {
            Some(val) => val,
            None => {
                // If value is not present in cache, get it from the storage.
                let key = Self::get_key_from_shard_uid_and_hash(self.shard_uid, hash);
                let val = self
                    .store
                    .get(ColState, key.as_ref())
                    .map_err(|_| StorageError::StorageInternalError)?
                    .ok_or_else(|| {
                        StorageError::StorageInconsistentState("Trie node missing".to_string())
                    })?;
                let val: Arc<[u8]> = val.into();

                // Insert value to shard and chunk caches.
                if val.len() < TRIE_LIMIT_CACHED_VALUE_SIZE {
                    self.shard_cache.put(*hash, val.clone());
                }
                val
            }
        };

        self.inc_counter();
        if let TrieCacheState::CachingChunk = this.cache_state.borrow().get() {
            self.chunk_cache.borrow_mut().insert(key, val);
        };

        Ok(val)
    }
```


One more thing I like about the above suggestion is that we move `self.inc_counter()` *after* fetcing from the DB. This is nice, because, if `.store.get` fails, we don't increment the counter (if we increment the counter but *dont* actually fetch the node, we break atomicity).